src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,591 @@
+
+        # data_version_filter_1 = [(0, Clusters.Descriptor, data_version_1)]
+
+        read_request_27_2_1 = await self.default_controller.ReadAttribute(self.dut_node_id, [(0, Clusters.BasicInformation.Attributes.NodeLabel)])
+        data_version_2 = read_request_27_2_1[0][Clusters.BasicInformation][Clusters.Attribute.DataVersion]
+        # data_version_filter_2 = [(0, Clusters.Descriptor, data_version_2)]
+
+        asserts.assert_not_equal(data_version_1, data_version_2)
+
+        # Step 28
+
+        # TH sends a Read Request Message to the DUT to read something(Attribute) which is larger than 1 MTU(1280 bytes) and per spec can be chunked +
+
+        # Verify on the TH that the DUT sends a chunked data message with the SuppressResponse field set to False for all the messages except the last one.
+        # Verify the last chunked message sent has the SuppressResponse field set to True.
+
+        # self.print_step(28, "Send the Read Request Message to read something(Attribute) which is larger than 1 MTU(1280 bytes) and per spec can be chunked +")
+        # This apparently already exists in TCP tests -- remove?
+
+        # Step 29
+
+        # TH sends a Read Request Message to the DUT to read a non global attribute from all clusters at that Endpoint
+        # AttributePath = [[Endpoint = Specific Endpoint, Attribute = Specific Non Global Attribute]] +
+
+        # On the TH verify that the DUT sends an error message and not the value of the attribute.
+        self.print_step(29, "Send the Read Request Message to the DUT to read a non global attribute from all clusters at that Endpoint")
+        found_non_global = False
+        for endpoint_id, endpoint in self.endpoints.items():
+            if not found_non_global:
+                # global_attribute_ids.AttributeIdType.kStandardNonGlobal seems to be non-existent in chip-all-clusters-app
+                # But kTest does exist -> Clusters.Objects.UnitTesting
+                for cluster_type, cluster in endpoint.items():
+
+                    if global_attribute_ids.cluster_id_type(cluster_type.id) != global_attribute_ids.ClusterIdType.kStandard:
+                        # if global_attribute_ids.cluster_id_type(cluster_type.id) == global_attribute_ids.AttributeIdType.kStandardNonGlobal:
+                        found_non_global = True
+
+                        non_global_attr = list(ClusterObjects.ALL_ATTRIBUTES[cluster_type.id].values())[0]
+                        output = await self.default_controller.ReadAttribute(self.dut_node_id, [(0, non_global_attr)])
+                        asserts.assert_true(isinstance(output[0][cluster_type][non_global_attr].Reason, InteractionModelError),
+                                            msg="Unexpected success reading non-global attribute")
+                        continue
+
+        # Step 30
+
+        # TH sends a Read Request Message to the DUT to read a non global attribute from all clusters at all Endpoints
```

Comment:
same comment as last step

Reply:
Hardcoded to use Clusters.Objects.LaundryWasherControls